charts/skypilot/values.yaml

```diff
@@ -76,23 +76,45 @@ ingress:
   authCredentials: "username:$apr1$encrypted_password"
   # The base path of API server, you may use differenet path to expose multiple API server through a unified ingress-controller
   path: '/'
-  # GKE load balancer used for ingress does not work with websocket connections
-  # (required for our k8s SSH tunneling), in such case NodePort should be enabled
-  # to exposes the ingress controller.
-  # Note: nodePort svc will only be created when both ingress.nodePortEnabled=true AND ingress-nginx.enabled=true
-  nodePortEnabled: true
+
+  # TODO(aylei): document the migration steps in v0.9.0 release note
+  # Whether enable an additional NodePort service for the ingress controller
+  # Default to null for backward compatibility:
+  # - For existing deployments, keep the service as is so that upgrade does not break existing NodePort endpoint (if any)
+  # - For new deployments, disable the NodePort service, equivalent to ingress.nodePortEnabled=false
+  # So, for deployments that use default values, upgrade from 0.8.0 nightly will:
+  # - Keep the existing NodePort endpoint
+  # - Create a new LoadBalancer endpoint since ingress-nginx.controller.service.type is now default to LoadBalancer
+  # It is recommended to keep only one endpoint by explicitly setting ingress.nodePortEnabled or ingress-nginx.controller.service.type
+  # if you are upgrading from 0.8.0 nightly.
+  # Deprecated: use ingress-nginx.controller.service.type=NodePort instead
+  nodePortEnabled: null
   # Specific nodePort to use for the ingress controller
   # If not set, Kubernetes will assign random ports in the NodePort range (default 30000-32767)
+  # Deprecated: use ingress-nginx.controller.service.nodePorts.http instead
```

Comment:
Can we add a note to remove this before 0.9.0 release? Same for the nodePortEnabled field.

Reply:
Added the note